Sources/ArcGISToolkit/Components/Augmented Reality/Utilities/ARSwiftUIView.swift

```diff
@@ -134,22 +135,41 @@ class ARSwiftUIViewProxy: NSObject, ARSessionProviding {
     @objc dynamic var session: ARSession {
         arView.session
     }
-    
-    /// Creates a raycast query that originates from a point on the view, aligned with the center of the camera's field of view.
-    /// - Parameters:
-    ///   - point: The point on the view to extend the raycast from.
-    ///   - target: The type of surface the raycast can interact with.
-    ///   - alignment: The target's alignment with respect to gravity.
-    /// - Returns: An `ARRaycastQuery`.
-    func raycastQuery(
-        from point: CGPoint,
-        allowing target: ARRaycastQuery.Target,
-        alignment: ARRaycastQuery.TargetAlignment
-    ) -> ARRaycastQuery? {
-        return arView.raycastQuery(
-            from: point,
-            allowing: target,
-            alignment: alignment
+}
+
+extension ARSwiftUIViewProxy {
+    /// Performs a raycast to get the transformation matrix representing the corresponding 
+    /// real-world point for `screenPoint`.
+    /// - Parameter screenPoint: The screen point to determine the real world transformation matrix from.
+    /// - Returns: A `TransformationMatrix` representing the real-world point corresponding to `screenPoint`.
+    func raycast(from screenPoint: CGPoint) -> TransformationMatrix? {
+        // Use the `raycastQuery` method on ARSCNView to get the location of `screenPoint`.
+        guard let query = arView.raycastQuery(
+            from: screenPoint,
+            allowing: .existingPlaneGeometry,
```

Comment:
I'm a little bit hesitant to make a change here. According to the doc, an [existingPlaneGeometry](https://developer.apple.com/documentation/arkit/arraycastquery/target/existingplanegeometry) is for "a plane to have a definitive size and shape", such as a table top. An [estimatedPlane](https://developer.apple.com/documentation/arkit/arraycastquery/target/estimatedplane) is for nonplanar surfaces, such as a real-world surface. 

I was tempted to change the method to take the allowed target plane as a parameter, so that table top AR can pass in `existingPlaneGeometry` and world scale AR can use `estimatedPlane`. Do you think that would be a good idea?

It may be hard to quantify how much effect it has on the raycasting accuracy in real world.

Reply:
> I was tempted to change the method to take the allowed target plane as a parameter, so that table top AR can pass in existingPlaneGeometry and world scale AR can use estimatedPlane. Do you think that would be a good idea?

Yes I agree that the allowed target plane should be a parameter. I tested both target types and .estimatedPlane is better for a real-world surface.